sdk/identity/azure-identity/azure/identity/aio/_credentials/environment.py

```diff
@@ -75,8 +75,10 @@ def __init__(self, **kwargs: Any) -> None:
             expected_variables = set(EnvironmentVariables.CERT_VARS + EnvironmentVariables.CLIENT_SECRET_VARS)
             set_variables = [v for v in expected_variables if v in os.environ]
             if set_variables:
-                _LOGGER.warning(
-                    "Incomplete environment configuration. These variables are set: %s", ", ".join(set_variables)
+                _LOGGER.log(
+                    logging.INFO if kwargs.get("_within_dac") else logging.WARNING,
```

Comment:
If it is within dac, we don't need to log anything, right?

It is not the case of "Incomplete environment configuration for EnvironmentCredential."